compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9973,6 +9991,74 @@ private STNode parseNamespacePrefix() {
         }
     }
 
+    private STNode parseClientDeclOrVarDeclStatement(STNode metadata, List<STNode> qualifiers) {
+        return parseClientDeclOrVarDecl(metadata, STNodeFactory.createEmptyNode(), qualifiers, false);
+    }
+
+    private STNode parseClientDeclOrVarDecl(STNode metadata, STNode publicQualifier, List<STNode> qualifiers,
+                                            boolean isModuleVar) {
+        switch (peek().kind) {
+            case STRING_LITERAL_TOKEN:
+                reportInvalidMetaData(metadata, "client declaration");
+                reportInvalidQualifier(publicQualifier);
+                return parseClientDeclaration(qualifiers.get(qualifiers.size() - 1), isModuleVar);
+            case OBJECT_KEYWORD:
+                if (isModuleVar) {
+                    return parseModuleVarDecl(metadata, publicQualifier, qualifiers);
+                }
+                return parseVariableDecl(getAnnotations(metadata), publicQualifier, new ArrayList<>(), qualifiers,
+                                         false);
+            default:
+                recover(peek(), ParserRuleContext.CLIENT_DECL_OR_CLIENT_OBJECT_VAR_DECL);
+                return parseClientDeclOrVarDecl(metadata, publicQualifier, qualifiers, isModuleVar);
+        }
+    }
+
+    /**
+     * Parse client declaration.
+     * <p>
+     * <code>client-decl := client client-decl-uri as client-decl-prefix;
+     * <br/>
+     * client-decl-uri := string-literal
+     * <br/>
+     * client-decl-prefix := identifier
+     * </code>
+     *
+     * @return client declaration node
+     */
+    private STNode parseClientDeclaration(STNode clientKeyword, boolean isModuleVar) {
+        startContext(ParserRuleContext.CLIENT_DECLARATION);
+
+        STNode clientDeclUri = parseStringLiteral();
+        STNode clientDecl = parseClientDeclWithAlias(clientKeyword, clientDeclUri, isModuleVar);
+        
+        endContext();
+        return clientDecl;
+    }
+
+    private STNode parseClientDeclWithAlias(STNode clientKeyword, STNode clientDeclUri, boolean isModuleVar) {
+
+        STNode asKeyword = parseAsKeyword();
+        STNode prefix = parseClientDeclPrefix();
```

Comment:
```suggestion
        STNode prefix = parseIdentifier(ParserRuleContext.IDENTIFIER);
```

Let's remove the `CLIENT_DECL_PREFIX` ctx